src/views/virtualmachines/search/hooks/useVirtualMachineSearchSuggestions.ts

```diff
@@ -0,0 +1,99 @@
+import { useMemo } from 'react';
+
+import {
+  VirtualMachineModelGroupVersionKind,
+  VirtualMachineInstanceModelGroupVersionKind,
+} from '@kubevirt-ui/kubevirt-api/console';
+import { V1VirtualMachine, V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { getName, getNamespace, getAnnotation, getLabels } from '@kubevirt-utils/resources/shared';
+import { getVMIIPAddresses } from '@kubevirt-utils/resources/vmi/utils/ips';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { SearchSuggestResult } from '@search/components/SearchSuggestBox';
+import { OBJECTS_FETCHING_LIMIT, compareCIDR } from '@virtualmachines/utils';
+
+type UseVirtualMachineSearchSuggestions = (
+  searchQuery: string,
+) => [result: SearchSuggestResult, loaded: boolean];
+
+export const useVirtualMachineSearchSuggestions: UseVirtualMachineSearchSuggestions = (
+  searchQuery,
+) => {
+  const [vms, vmsLoaded] = useK8sWatchResource<V1VirtualMachine[]>({
+    groupVersionKind: VirtualMachineModelGroupVersionKind,
+    isList: true,
+    limit: OBJECTS_FETCHING_LIMIT,
+    namespaced: true,
+  });
+
+  const [vmis, vmisLoaded] = useK8sWatchResource<V1VirtualMachineInstance[]>({
+    groupVersionKind: VirtualMachineInstanceModelGroupVersionKind,
+    isList: true,
+    limit: OBJECTS_FETCHING_LIMIT,
+    namespaced: true,
+  });
+
+  const vmsToSuggest = useMemo<V1VirtualMachine[]>(
+    () =>
+      vmsLoaded
+        ? vms.filter((vm) => getName(vm).toLowerCase().includes(searchQuery.toLowerCase()))
+        : [],
```

Comment:
probably
(vms??[]).filter((vm) => getName(vm).toLowerCase().includes(searchQuery.toLowerCase())) will work the same, u can check

Reply:
The intent was "only compute the value if VMs are loaded, otherwise return an empty array".

IMO an explicit `vmsLoaded` condition guarding the computation is more readable.